pkgs/development/python-modules/fmpy/default.nix

```
@@ -0,0 +1,215 @@
+  notebook,
+  plotly,
+  hatchling,
+  pyside6,
+  jinja2,
+  flask,
+  dash,
+  dash-bootstrap-components,
+  qt6,
+  fmpy,
+  runCommand,
+  enableRemoting ? false,
+}:
+
+let
+  # From sundials, build only the CVODE solver. C.f.
+  # src/native/build_cvode.py
+  cvode =
+    (sundials.overrideAttrs (prev: {
+      # hash copied from native/build_cvode.py
+      version = "5.3.0";
+      src = fetchurl {
+        url = "https://github.com/LLNL/sundials/releases/download/v5.3.0/sundials-5.3.0.tar.gz";
+        sha256 = "88dff7e11a366853d8afd5de05bf197a8129a804d9d4461fb64297f1ef89bca7";
+      };
+
+      cmakeFlags =
+        prev.cmakeFlags
+        ++ lib.mapAttrsToList (option: enable: lib.cmakeBool option enable) {
+          # only build the CVODE solver
+          BUILD_CVODE = true;
+
+          BUILD_CVODES = false;
+          BUILD_ARKODE = false;
+          BUILD_IDA = false;
+          BUILD_IDAS = false;
+          BUILD_KINSOL = false;
+        };
+    })).override
+      {
+        lapackSupport = false;
+        lapack.isILP64 = stdenv.hostPlatform.is64bit;
+        blas = lapack;
+        kluSupport = false;
+      };
+
+in
+buildPythonPackage rec {
+  pname = "fmpy";
+  version = "0.3.23";
+  disabled = pythonOlder "3.10";
+  pyproject = true;
+
+  # Bumping version? Make sure to look through the commit history for
+  # bumped native/build_cvode.py pins (see above).
+  src = fetchFromGitHub {
+    owner = "CATIA-Systems";
+    repo = "FMPy";
+    rev = "v${version}";
+    fetchSubmodules = true;
+    hash = "sha256-4jBYOymurGCbjT0WQjIRNsztwOXBYVTGLdc4kNSTOZw=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    hatchling
+  ];
+
+  propagatedBuildInputs = [
+    pyqtgraph
+    pyside6
+    attrs
+    lark
+    lxml
+    msgpack
+    numpy
+    scipy
+    pytz
+    dask
+    requests
+    matplotlib
+    pyqtgraph
+    notebook
+    plotly
+    rpclib
+    cvode
+    pyside6
+    jinja2
+    flask
+    dash
+    dash-bootstrap-components
+  ];
+
+  dontUseCmakeConfigure = true;
+  dontUseCmakeBuildDir = true;
+
+  patches = [
+    ./0001-gui-override-Qt6-libexec-path.patch
+  ];
+
+  postPatch =
+    ''
+      substitute ${./libraries.py} ./src/fmpy/sundials/libraries.py \
+          --subst-var-by cvode ${cvode}
+
+      # Sundials are provided outside of this package
+      sed --regexp-extended --in-place '/(sundials_)/d' pyproject.toml
+    ''
+    # Remove artifacts that we do not build
+    + lib.optionalString stdenv.isLinux ''
+      sed --regexp-extended --in-place '/(.dll|.dylib|.exe)/d' pyproject.toml
+    ''
+    + lib.optionalString stdenv.isDarwin ''
+      sed --regexp-extended --in-place '/(.dll|.so|linux|.exe)/d' pyproject.toml
+    ''
+    + lib.optionalString (!enableRemoting) ''
+      sed --regexp-extended --in-place '/(client_tcp|server_tcp)/d' pyproject.toml
+    '';
+
+  # Don't run upstream build scripts as they are too specialized.
+  # cvode is already built, so we only need to build native binaries.
```

Another worth noting thing to comment about could be:

```suggestion
# cvode is already built, so we only need to build native binaries.
# We run these cmake builds and then run the standard Nixpkgs Python
# setup-hooks.
```

---

This review comment is not addressed.

---

We are not using a nixpkgs-provided setup-hook. Do we comment that we use the vendored Python-function instead, perhaps?

---

> We are not using a nixpkgs-provided setup-hook. Do we comment that we use the vendored Python-function instead, perhaps?

Hmm well indeed maybe my comment suggestion wasn't clear enough :) Maybe this is better:

```suggestion
  # cvode is already built, so we only need to build native binaries.
  # We run these cmake builds and then run the standard buildPythonPackage
  # phases.
```